From bf3dcae3c4ea61ff4cb68e1cab8210dd38d7859a Mon Sep 17 00:00:00 2001
From: Konrad Leszczynski <konrad.leszczynski@intel.com>
Date: Wed, 9 Dec 2015 14:49:05 +0000
Subject: [PATCH 2/4] usb: ssic: fix the obsolete suspend call in the sideband
 wakeup

This patch removes the unnecessary call of usb_autosuspend_device
from the sideband wakeup driver. If the usb_autoresume_device
call will be unsuccessful, the function itself will re-increment
the device usage counter.

Change-Id: Ia8ae426e7802ae2b614c66d4a2b4824c8a68b603
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-11836
Signed-off-by: Konrad Leszczynski <konrad.leszczynski@intel.com>
Reviewed-on: https://android.intel.com:443/446286
---
 drivers/platform/x86/intel-gpio-ssic.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/drivers/platform/x86/intel-gpio-ssic.c b/drivers/platform/x86/intel-gpio-ssic.c
index 3b7192a..0338c05 100644
--- a/drivers/platform/x86/intel-gpio-ssic.c
+++ b/drivers/platform/x86/intel-gpio-ssic.c
@@ -101,8 +101,7 @@ static irqreturn_t gpio_ssic_wakeup_irq(int irq, void *__ssic)
 	}
 
 	usb_lock_device(udev);
-	if (!usb_autoresume_device(udev))
-		usb_autosuspend_device(udev);
+	usb_autoresume_device(udev);
 	usb_unlock_device(udev);
 	usb_put_dev(udev);
 
-- 
1.9.1

